kernel/mutex.c

```diff
@@ -42,7 +43,7 @@ int __myst_mutex_trylock(myst_mutex_t* m, myst_thread_t* self)
     if (m->owner == NULL)
     {
         /* If the waiters queue is empty */
-        if (m->queue.front == NULL)
```

Comment:
All changes to mutex.h are unecessary and were introduced by the accessors for condition. Mutexes and conditions are tightly coupled intentionally and mutex should not be considered a consumer of condition.

Reply:
`kernel/mutex.c` has been reverted.